olca-app/src/org/openlca/app/editors/graphical/model/ProcessNode.java

```diff
@@ -226,6 +231,46 @@ public List<ExchangeNode> loadExchangeNodes() {
 		return getExchangeNodes();
 	}
 
+	public boolean shouldDisplayInputs() {
+		if (process == null)
+			return false;
+		if (process instanceof ProcessDescriptor p) {
+			return parent().isWasteProcess(this)
+				|| (!p.isFromLibrary() && p.processType == ProcessType.UNIT_PROCESS);
+		}
+		else return false;
+	}
+
+	public boolean shouldDisplayOutputs() {
```

Comment:
This is not correct, as this would not display the output of a sub-system, library process etc. Also, the method above is not correct for waste flows of sub-systems etc. Also, would be better if the display related things do not go into the model.

Reply:
Yes, I forgot that those ones are not under `ProcessDescriptor`.

I initially started by putting everything into `IOFigure`. But, the issue is that `IONode` will create `ExchangeNode`s via [`addExchanges`](https://github.com/GreenDelta/olca-app/blob/e20b93e8fe81ed0f3076150dba718411c9167a8c/olca-app/src/org/openlca/app/editors/graphical/model/IONode.java#L39) that won't be used in the graph display. So, I thought that we should not create those `ExchangeNode`s if they are not displayed in the graph. That is the reason why, I am filtering it in the [`IONode` model](https://github.com/GreenDelta/olca-app/blob/2f84e7abae0b7f9e4de63ca0d7a828c0a4089314/olca-app/src/org/openlca/app/editors/graphical/model/IONode.java#L52-L56) using `shouldDisplay...` in the model too.

So, questions:
Don't you think it is suboptimal to create `Exchange` objects that won't be used?